payments/swish/seamless-view.md

```diff
@@ -18,120 +18,386 @@ sidebar:
+The Seamless View integration provide you with the Swish payment solution
+directly on your website. This gives the consumer a frictionless experience as
+we are handling the payment in the implemented `iframe` on your page (see
+example below).
+
+![screenshot of the seamless view swish payment
+page][seamless-view-img]{:height="250px" width="660px"}
+
+## Purchase flow
+
+```mermaid
+sequenceDiagram
+    participant Payer
+    participant Merchant
+    participant SwedbankPay as Swedbank Pay
+
+    activate Payer
+    Payer->>-Merchant: start purchase
+    activate Merchant
+    note left of Payer: First API request
+    Merchant->>-SwedbankPay: POST /psp/swish/payments ①
+    activate SwedbankPay
+    SwedbankPay-->>-Merchant: rel: view-sales ②
+    activate Merchant
+    Merchant-->>-Payer: authorization page
+    activate Payer
+    note left of Payer: Open iframe ③
+    Payer->>Payer: Input mobile number
+    Payer->>-SwedbankPay: Show Consumer UI page in iframe - Authorization ④
+    activate SwedbankPay
+        opt Card supports 3-D Secure
+        SwedbankPay-->>-Payer: redirect to IssuingBank
+        activate Payer
+        Payer->>IssuingBank: 3-D Secure authentication process
+        activate IssuingBank
+        IssuingBank->>-Payer: 3-D Secure authentication process
+        Payer->>-SwedbankPay: access authentication page
+        end
+    SwedbankPay-->>Merchant: Event: OnPaymentComplete ⑤
+    activate Merchant
+    note left of Merchant: Second API request.
+    Merchant->>-SwedbankPay: GET <payment.id> ⑥
+    activate SwedbankPay
+    SwedbankPay-->>-Merchant: rel: view-payment
+    activate Merchant
+    Merchant-->>-Payer: display purchase result
+    activate Payer
+
+        opt Callback is set ⑦
+        activate SwedbankPay
+        SwedbankPay->>SwedbankPay: Payment is updated
+        SwedbankPay->>-Merchant: POST Payment Callback
+        end
+```
+
+### Explainations
+
+* ① When the payer starts the purchase process, you make a `POST` request
+  towards Swedbank Pay with the collected Purchase information.
+* ② `rel: view-sales` is a value in one of the operations, sent as a response
+  from Swedbank Pay to the Merchant.
+* ③ `Open iframe` creates the Swedbank Pay hosted iframe.
+* ④ `Show Consumer UI page in iframe` displays the payment window as content
+  inside of the iframe. The consumer can insert mobile information for
+  authorization.
+* ⑤ `Event: OnPaymentComplete` is when er payment is complete. Please note that
+  both a successful and rejected payment reach completion, in contrast to a
+  cancelled payment.
+* ⑥ To get the transaction result, you need to follow up with a `GET` request
+  using the `paymentID` received in the first step.
+* ⑦ If CallbackURL is set you will receive a payment callback when the Swish
   dialogue is completed, and you will have to make a `GET` request to check the
   payment status.
-* The flow is explained in the sequence diagram below.
 
-## API Requests
+### 3-D Secure
```

Comment:
As for Vipps, I don't think 3-D Secure is relevant for Swish payments. Perhaps when you register for Swish the first time, but not when performing payments.